src/linux/system.rs

```diff
@@ -248,32 +248,26 @@ impl SystemExt for System {
     }
 
     fn refresh_memory(&mut self) {
-        if let Ok(data) = get_all_data("/proc/meminfo", 16_385) {
-            let mut mem_available_found = false;
-
-            for line in data.split('\n') {
-                let mut iter = line.split(':');
-                let field = match iter.next() {
-                    Some("MemTotal") => &mut self.mem_total,
-                    Some("MemFree") => &mut self.mem_free,
-                    Some("MemAvailable") => {
+        let mut mem_available_found = false;
+        if let Some(meminfo_table) = read_table("/proc/meminfo", ':') {
+            for (key, value_kib) in meminfo_table {
+                // /proc/meminfo reports KiB, though it says "kB". Convert it.
+                let value = value_kib.saturating_mul(1_024);
```

Comment:
Before your change, we only did this operation if there was an associated field. I'd prefer to keep it the previous way if you don't mind.

Reply:
It will make the code more verbose because the `saturating_mul` will be repeated a lot. Do you still prefer that?

Reply:
I'm not super happy with the current state of the code. I see two other possibilies:
 1. A function marked as `#[inline]` taking a `&mut u64` and doing that.
 2. A macro.

If you have another suggestion, it's good too. :)

Reply:
I tend to favor performance over readability. In this case readability could be improved without impacting performance though. The current code is especially ugly imo (mine, not your changes).